src/graphql/ip_location.rs

```diff
@@ -37,6 +38,46 @@ impl IpLocationQuery {
 
         Ok(record.transpose()?)
     }
+
+    /// The list of locations for up to 200 IP addresses.
+    #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
+        .or(RoleGuard::new(Role::SecurityAdministrator))
+        .or(RoleGuard::new(Role::SecurityManager))
+        .or(RoleGuard::new(Role::SecurityMonitor))")]
+    async fn ip_location_list(
+        &self,
+        ctx: &Context<'_>,
+        mut addresses: Vec<String>,
+    ) -> Result<Vec<IpLocationItem>> {
+        let Ok(mutex) = ctx.data::<Arc<Mutex<ip2location::DB>>>() else {
+            return Err("IP location database unavailable".into());
+        };
+
+        addresses.truncate(MAX_NUM_IP_LOCATION_LIST);
+        let records = if let Ok(mut locator) = mutex.lock() {
+            addresses
+                .iter()
+                .filter_map(|address| {
+                    address.parse::<IpAddr>().ok().and_then(|addr| {
+                        locator
+                            .ip_lookup(addr)
+                            .ok()
+                            .map(std::convert::TryInto::try_into)
+                            .and_then(|r| {
+                                r.ok().map(|location| IpLocationItem {
+                                    address: address.clone(),
+                                    location,
+                                })
+                            })
+                    })
+                })
+                .collect()
+        } else {
+            Vec::new()
```

Comment:
If `Mutex::lock` fails, an empty vector is returned. However, according to the [docs](https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.lock), `Mutex::lock` only fails if another thread panicked while holding the mutex. In this case, I think it would be better to return an error instead of an empty vector, as it's not a normal or expected result.

Reply:
Thank you for your suggestion. Done.